root/ntuple/makeproject/rntuple/CMakeLists.txt

```
@@ -0,0 +1,75 @@
+if(NOT ROOT_root7_FOUND)
+  return()
+endif()
+
+# Generate common dictionary for custom class
+ROOTTEST_GENERATE_DICTIONARY(
+  rntuple_event_dict
+  ${CMAKE_CURRENT_SOURCE_DIR}/ntuple_makeproject_header.h
+  LINKDEF ${CMAKE_CURRENT_SOURCE_DIR}/ntuple_makeproject_header_LinkDef.h
+  # Otherwise TFile::MakeProject will find the dictionary module and will not generate the corresponding class header
+  NO_ROOTMAP NO_CXXMODULE
+  FIXTURES_SETUP generated_event_dictionary_rntuple
+)
+
+# Write the RNTuple with the custom class field
+ROOTTEST_GENERATE_EXECUTABLE(
+  write_rntuple
+  LIBRARIES Core RIO ROOTNTuple
+  FIXTURES_REQUIRED generated_event_dictionary_rntuple
+  FIXTURES_SETUP write_rntuple_executable)
+# Specify more than one input source for the generated executable
+target_sources(
+  write_rntuple
+  PRIVATE write_rntuple.cxx rntuple_event_dict.cxx
+)
+
+ROOTTEST_ADD_TEST(write_rntuple
+                  EXEC ./write_rntuple
+                  FIXTURES_REQUIRED write_rntuple_executable
+                  FIXTURES_SETUP written_rntuple)
+
+# Call MakeProject on the output file
+ROOTTEST_GENERATE_EXECUTABLE(
+  makeproject_rntuple
+  makeproject_rntuple.cxx
+  LIBRARIES Core RIO ROOTNTuple
+  FIXTURES_SETUP makeproject_rntuple)
+ROOTTEST_ADD_TEST(makeproject_rntuple
+                  EXEC ./makeproject_rntuple
+                  FIXTURES_REQUIRED "written_rntuple;makeproject_rntuple"
+                  FIXTURES_SETUP makeproject_rntuple_called)
+
+# Read back the class instance thanks to the shared library generated by MakeProject.
+# Make sure it corresponds to the data stored in the original custom class
+if(MSVC)
+  # Windows is more picky with the library name, provide full relative path to the test lib
```

```suggestion
  # Windows is more picky with the library name, provide full path to the test lib
```
isn't `CMAKE_CURRENT_BINARY_DIR` a full path?